Feat disable component when node is not built

EstherDarkish opened the pull request:
Disables all the interface components when the currentNode is disabled.

SlimaneAmar commented:
We have already `isNodeValid` function whichseems to be same than `isNodeDisabled` function ?
A node (model) is invalid and not disabled (component)

AbdelHedhili commented on src/components/graph/util/model-functions.js:
```diff
@@ -30,9 +30,13 @@ export function getFirstNodeOfType(elements, nodeType, buildStatus) {
+}
 
-    return !readOnlyNode && builtCurrentNode;
+export function isNodeDisabled(node) {
```
as said by @SlimaneAmar rename to isNodeInvalid

AbdelHedhili commented on src/components/study-container.js:
```diff
@@ -339,9 +345,9 @@ export function StudyContainer({ view, onChangeTab }) {
     }, [studyUuid, loadTree]);
 
     useEffect(() => {
-        if (!isNodeValid(currentNode) && currentNode?.type !== 'ROOT') return;
         loadNetwork(currentNode?.id === currentNodeIdRef.current);
```
why are we not checking if the node is built anymore ?

AbdelHedhili replied:
yes indeed no need then

EstherDarkish commented on src/components/network/equipment-table.js:
```diff
@@ -77,6 +77,7 @@ export const EquipmentTable = (props) => {
             )}
             {props.fetched &&
                 props.rows &&
```
Yes ok. And also for `props.columns?.length > 0`.

AbdelHedhili commented on src/components/app-top-bar.js:
```diff
@@ -242,7 +242,8 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
                 onEquipmentLabellingClick={handleChangeUseName}
                 equipmentLabelling={useNameLocal}
-                withElementsSearch={Boolean(studyUuid)}
+                withElementsSearch={true}
```
why this change ?

JacquesBorsenberger replied:
as the search function can now be disabled it prevent it from disappearing

AbdelHedhili commented on src/components/run-button-container.js:
```diff
@@ -16,7 +16,7 @@ import makeStyles from '@mui/styles/makeStyles';
 import { useDispatch, useSelector } from 'react-redux';
 import { useIntl } from 'react-intl';
-import { isNodeValid } from './graph/util/model-functions';
+import {} from './graph/util/model-functions';
```
remove

EstherDarkish replied:
Nice catch

FranckLecuyer reviewed (changes requested):
Tests:

. Sometimes, the 'LOG' component is totally empty (no disabled 'Log single node' button and no alert message 'Invalid Node')
. When an empty SLD is shown on an invalid node, and if we switch to 'SPREADSHEET' and return to 'MAP' view, the SLD is not shown anymore
. On an invalid node, the search bar is not disabled

EstherDarkish commented:
> Tests:
> 
> . Sometimes, the 'LOG' component is totally empty (no disabled 'Log single node' button and no alert message 'Invalid Node') 

Fixed

> . When an empty SLD is shown on an invalid node, and if we switch to 'SPREADSHEET' and return to 'MAP' view, the SLD is not shown anymore 

This is not a trivial task. @JacquesBorsenberger can you help me with this ? When single-line-diagram-pane.js::284 is commented, we have the old Voltage Level error back.

> . On an invalid node, the search bar is not disabled

Fixed
